src/mats_l1_processing/instrument.py

```diff
@@ -324,18 +324,23 @@ def __init__(self, channel, calibration_file):
 
         # single event correction
         filename = calibration_data['hot_pixels']['single_events']
-        file_conn = sqlite.connect(filename)
-        self.single_event = sqlite.connect(':memory:')
-        file_conn.backup(self.single_event)
-        file_conn.close()
-
+        if MATS_SQLITE_S3:
```

Comment:
Does not really matter, but you could move the logic to "sqlite_s3" and let that module work as a "gateway" - returning the single_event data (regardless where it came from).

Reply:
That might be neater – would hide the magic from the main code. I'll have a look with future possible patterns in mind.